types/framework/entity.d.ts

```diff
@@ -440,7 +443,8 @@ declare class Entity<D extends Entity.Data = Entity.Data> {
    * @param id           - The numeric ID of the child to retrieve
    * @param strict       - Throw an Error if the requested id does not exist, otherwise return null. Default false.
    */
-  getEmbeddedEntity(embeddedName: string, id: string, { strict }?: { strict?: boolean }): any;
+  getEmbeddedEntity<T extends keyof E>(embeddedName: T, id: string, { strict }: { strict: true }): E[T];
+  getEmbeddedEntity<T extends keyof E>(embeddedName: T, id: string, { strict }?: { strict?: boolean }): E[T] | null;
```

Comment:
making this `{ strict?: false }` would be less ambiguous I think.

Reply:
it doesn't actually correctly infer it in the case where you provide a boolean that's not known at compile time then:
```ts
item.getEmbeddedEntity('ActiveEffect', 'someId', {strict: (): boolean => true}); // error, doesn't match any overload
```
so I'd need to provide 3 overloads, one for `true`, one for `false`, and one for `boolean`. Because the overloads for `false` and `boolean` are identical, I just omitted the one for `false`.

Reply:
Okay. That makes sense. :+1: